packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import type { MongoClientOptions } from 'mongodb';
+import {
+  spacing,
+  Label,
+  Description,
+  Link,
+  Table,
+  TableHeader,
+  Row,
+  Cell,
+  Button,
+  IconButton,
+  Icon,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { editableUrlOptions, UrlOption } from '../../../utils/url-options';
+
+import UrlOptionsModal from './url-options-modal';
+
+const urlOptionsContainerStyles = css({
+  marginTop: spacing[3],
+  width: '70%',
+});
+
+const addUrlOptionsButtonStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[3],
+});
+
+const optionValueStyles = css({
+  span: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    width: '100%',
+  },
+});
+
+function UrlOptions({
+  handleFieldChanged,
+  connectionStringUrl,
+}: {
+  handleFieldChanged: (key: keyof MongoClientOptions, value: unknown) => void;
+  connectionStringUrl: ConnectionStringUrl;
+}): React.ReactElement {
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const [option, setOption] = React.useState<UrlOption | undefined>(undefined);
+
+  const urlOptions: UrlOption[] = [];
+  editableUrlOptions.forEach(({ values }) => {
+    values.forEach((name: string) => {
+      if (connectionStringUrl.searchParams.has(name)) {
+        urlOptions.push({
+          name: name as UrlOption['name'],
+          value: connectionStringUrl.searchParams.get(name) as string,
+        });
+      }
+    });
+  });
+
+  const addUrlOption = () => {
+    setOption(undefined);
+    setIsModalOpen(true);
+  };
+
+  const saveUrlOption = (option: UrlOption) => {
+    handleFieldChanged(option.name, option.value);
+    setOption(undefined);
+    setIsModalOpen(false);
+  };
+
+  const editUrlOption = (option: UrlOption) => {
+    setOption(option);
+    setIsModalOpen(true);
+  };
+
+  const deleteUrlOption = (name: UrlOption['name']) => {
+    handleFieldChanged(name, '');
+  };
+
+  return (
+    <div className={urlOptionsContainerStyles} data-testid="url-options">
+      <Label htmlFor={''}>Url Options</Label>
+      <Description>
+        Add additional MongoDB url options to customize your connection.&nbsp;
+        <Link
+          href={
+            'https://docs.mongodb.com/manual/reference/connection-string/#connection-string-options'
+          }
+        >
+          Learn More
+        </Link>
+      </Description>
+      {urlOptions.length > 0 && (
+        <Table
```

Comment:
Sorry if this was discussed before, but did we consider other ways of doing this type of input? Seems a bit cumbersome to have modal pop up every time you need to add / edit the value (there is also seems to be no way to delete it right now?)

Maybe we can use the host input pattern that allows to add / remove items with + / - button and just have two inputs (select for the key and text input for the value) for every line there?